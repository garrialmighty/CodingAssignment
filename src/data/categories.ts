import {Category} from 'src/redux/reducer/types';

export const DefaultCategories: Category[] = [
  {
    name: 'Cities',
    items: [
      {
        description: 'Capital of Singapore',
        answer: 'SINGAPORE',
        points: 10,
      },
      {
        description: 'Capital of France',
        answer: 'PARIS',
        points: 6,
      },
    ],
  },
  {
    name: 'Food',
    items: [
      {
        description: 'A spicy noodle dish popular in Southeast Asia.',
        answer: 'LAKSA',
        points: 6,
      },
    ],
  },
  {
    name: 'Animals',
    items: [
      {
        description: 'Insects that have large, often, brightly coloured wings.',
        answer: 'BUTTERFLY',
        points: 10,
      },
    ],
  },
  {
    name: 'Movies',
    items: [
      {
        description:
          '2012 American superhero film based on the Marvel comics of the same name.',
        answer: 'AVENGERS',
        points: 9,
      },
    ],
  },
];
