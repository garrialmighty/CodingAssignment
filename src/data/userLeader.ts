import {Leader} from 'src/redux/reducer/types';

export const UserLeader: Leader = {
  points: 0,
  position: 1,
  name: 'Current User',
};
