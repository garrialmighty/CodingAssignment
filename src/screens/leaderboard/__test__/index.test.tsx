import React from 'react';
import {Share} from 'react-native';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {act, fireEvent, render} from '@testing-library/react-native';

import {UserLeader} from 'src/data/userLeader';

import LeaderboardScreen from '../';

const mockStore = configureStore();
const state = {
  leader: {
    leaders: [UserLeader],
  },
};
const store = mockStore(state);

describe('Screen: Leaderboard', () => {
  it('should render properly', () => {
    const snapshot = render(
      <Provider store={store}>
        <LeaderboardScreen />
      </Provider>,
    ).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('can share points', () => {
    const mockShare = jest.fn();
    Share.share = mockShare;
    const {getByText} = render(
      <Provider store={store}>
        <LeaderboardScreen />
      </Provider>,
    );
    act(async () => await fireEvent.press(getByText('Share')));
    expect(mockShare).toHaveBeenCalled();
  });
});
