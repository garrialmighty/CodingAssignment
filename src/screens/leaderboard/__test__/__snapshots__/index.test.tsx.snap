// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen: Leaderboard should render properly 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "justifyContent": "space-between",
      "marginHorizontal": "5%",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 30,
      }
    }
  >
    1. Current User 	 - 	 0
  </Text>
</RCTSafeAreaView>
`;
